### PR TITLE
Answer input

### PR DESCRIPTION
**Pull request by bumblebot1**

### CODE REVIEW

File: src/client/game/answer-input.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react';
+
+import style from '../index.scss';
+
+const SPACE = 32
```
Comment (thullSL) [inline src/client/game/answer-input.js]: SPACE_KEY?

File: src/client/game/answer-input.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        });
+    }
+
+    handleEnterPress(event) {
+        switch (event.which) {
+            case ENTER_KEY: { //Enter key press
+                if(event.target.value.length > 0) {
+                    let answerList = this.state.answerList;
+                    for(let i = 0; i < answerList.length; i++) {
+                        console.log(answerList[i])
```
Comment (thullSL) [inline src/client/game/answer-input.js]: no sure if console logs add much business value to the production system.

**Comment by thullSL** [comment]
apart from the comments I've made :shipit:

File: src/client/game/answer-input.js (hunk 2 above)
Comment (JamieZafar) [inline src/client/game/answer-input.js]: https://memecrunch.com/meme/3OISI/alright-then/image.jpg?w=430&c=1

**Comment by JamieZafar** [comment]
:ship:

**Comment by JamieZafar** [comment]
is it still 2 :shipit: s?

**Comment by JamieZafar** [comment]
b4 we cn murj

**Comment by thullSL** [comment]
is it blocking?

**Comment by JamieZafar** [comment]
nuuu

**Comment by thullSL** [comment]
then calm yourself and wait for some more eyes.

**Comment by bumblebot1** [comment]
roasted

File: src/client/game/answer-input.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+
+import style from '../index.scss';
+
+const SPACE_KEY = 32
+  , ENTER_KEY = 13
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: I know the po asked you to do this, but I would probably just type out const a few more times :smile:

File: src/client/game/answer-input.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                })
+        });
+    }
+    handleFocus(index) {
+        this.setState({
+            focusIndex: index
+        });
+    }
+
+    textBoxGenerator(props, focusIndex) {
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: props? Should be answer list I suspect. Also - both these values are coming from the state and as we're in the component that's accessible in here. Use `const { answerList, focusIndex } = this.state;` and then you can remove these parameters

File: src/client/game/game.js [hunk 5]
```diff
@@ -1,9 +1,15 @@
 import React, { Component } from 'react';
 
-export default function Game(props) {
-    return (
-        <div className="text-center">
-            <h3>THIS IS GAME</h3>
-        </div>
-    );
+import style from '../index.scss';
```
Comment (kenneth-gray) [inline src/client/game/game.js]: Might be time to rename index.scss to common.scss, looking weird here. Index should be used in index.js or app.js and should be used to do general styling. Classes and ids should be moved to this common.scss :+1:

File: src/client/index.scss [hunk 6]
```diff
@@ -1,31 +1,39 @@
-h1 {
+.header1 {
```
Comment (kenneth-gray) [inline src/client/index.scss]: Why was this changed? Surely the site has a theming that means all h1 is this colour and therefore it would be a massive pain to have to import this every time I use an h1? I'm sure there's a good reason so let's discuss tomorrow :smile_cat:

File: src/client/game/answer-input.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        this.refs[focusIndex].focus();
+    }
+
+    render() {
+        return (
+            <div className="col-lg-12 text-center">
+                <h3 className={style.header3}>ANSWER INPUT</h3>
+                <button className="btn btn-primary btn-success">Ready</button>
+                <div>
+                    {this.textBoxGenerator(this.state.answerList, this.state.focusIndex)}
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: Not convinced this needed to be pulled out into a separate method in the class. It could have been done above the return and assigned to `const textBox`, then here would just be `<div>{textBox}</div>`. Thoughts?

File: src/client/game/answer-input.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    componentWillMount() {
+        this.setState({
+            answerList: [''],
+            focusIndex: 0
+        });
+    }
+
+    handleEnterPress(event) {
+        switch (event.which) {
+            case ENTER_KEY: { //Enter key press
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: You can remove the comment now the const exists

**Comment by thullSL** [comment]
:shipit:  again

**Comment by kenneth-gray** [comment]
:shipit:
